src/osd/PGLog.h

```diff
@@ -713,9 +724,10 @@ struct PGLog : DoutPrefixProvider {
   void roll_forward_to(
     eversion_t roll_forward_to,
     LogEntryHandler *h) {
-    log.roll_forward_to(
-      roll_forward_to,
-      h);
+    if (log.roll_forward_to(
+	  roll_forward_to,
+	  h))
+      dirty_log = false;
```

Comment:
Wouldn't this be diirty_log = true?  Also, like I said, probably easier to just set it in advance_can_rollback_to.

Reply:
Oh, sorry.

Reply:
Does this still need to be switched?

Reply:
@Aran85 I think @athanatos's comment makes sense, was there a reason we did not make that change or did you just not get to it?

Reply:
@Aran85 no worries, please create another PR for it and link http://tracker.ceph.com/issues/40403, we can backport both the fixes together.

Reply:
thanks so much!